add-on/src/lib/ipfs-path.js

```diff
@@ -38,6 +38,17 @@ function pathAtHttpGateway (path, gatewayUrl) {
 }
 exports.pathAtHttpGateway = pathAtHttpGateway
 
+function redirectSubdomainGateway (url, subdomainGateway) {
+  if (typeof url === 'string') {
+    url = new URL(url)
+  }
+  const fqdn = url.hostname.split('.')
+  const cid = fqdn[0]
+  const protocol = fqdn[1]
```

Comment:
I realized this won't produce correct results for gateways under more complex top level domains such as `cid.ipfs.something.co.uk`.

The fix is to switch to `is-ipfs`, probably something like:

```js
const { subdomainPattern } = require('is-ipfs')

const match = fqdn.match(subdomainPattern)
if (!match) throw new Error('no match for isIPFS.subdomainPattern')
const protocol = match[1]
const cid = match[2]
```

We should avoid `split('.')` and use `subdomainPattern`  here and in already existing `subdomainToIpfsPath` method.